runtime/wasm/src/module/mod.rs

```diff
@@ -124,11 +127,11 @@ pub(crate) struct WasmiModule {
     timeout_checkpoint_count: u64,
 }
 
-impl WasmiModule {
+impl<'a> WasmiModule<'a> {
```

Comment:
For `wasmtime` keeping this static is going to be a requirement.

The reason is that the current `wasmtime` API requires host exports to implement [IntoFunc](https://docs.rs/wasmtime/0.16.0/wasmtime/trait.IntoFunc.html), which is implemented only for static closures. Host exports need this of course so so I'll need to pass it into the closure, probably as an `Rc<RefCell<WasmiModule>>`.

Reply:
I pushed a change to make this `'static`. Not super happy about the change, since it turns a compile-time error into a runtime error. There wasn't a clear alternative though. Please update your review.